firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -124,6 +134,51 @@ public Thread newThread(Runnable r) {
     this.utils = utils;
     this.iidStore = iidStore;
     this.fidGenerator = fidGenerator;
+    preventParallelUsageOfIncompatibleVersionOfInstanceIdSDk();
+  }
+
+  private void preventParallelUsageOfIncompatibleVersionOfInstanceIdSDk() {
+    if (!isIIDVersionCompatible()) {
+      throw new IllegalStateException(
+          "FirebaseInstallations will not work correctly with current version of Firebase "
+              + "Instance ID SDK. Please update your Firebase Instance ID version.");
```

Comment:
These version of FirebaseInstanceId and FirebaseInstallations can not be used in parallel.
Please update your Firebase Instance ID to a version 20.1.0 or later.